sdk/dotnet/Pulumi.Automation/XStack.cs

```diff
@@ -0,0 +1,631 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Pulumi.Automation.Commands;
+using Pulumi.Automation.Commands.Exceptions;
+using Pulumi.Automation.Serialization;
+
+namespace Pulumi.Automation
+{
+    /// <summary>
+    /// <see cref="XStack"/> is an isolated, independently configurable instance of a
+    /// Pulumi program. <see cref="XStack"/> exposes methods for the full pulumi lifecycle
+    /// (up/preview/refresh/destroy), as well as managing configuration.
+    /// <para/>
+    /// Multiple stacks are commonly used to denote different phases of development
+    /// (such as development, staging, and production) or feature branches (such as
+    /// feature-x-dev, jane-feature-x-dev).
+    /// <para/>
+    /// Will dispose the <see cref="Workspace"/> on <see cref="Dispose"/>.
+    /// </summary>
+    public sealed class XStack : IDisposable // TODO: come up with a name for this
```

Comment:
I can't think of anything else that would make sense given that we are committed to `Stack` in the other SDKs. I think this could be `Pulumi.Automation.Stack` but yea, since `Stack` implements `Resource` it's possible we have collisions if they are used in the same program.

Reply:
`AutomationStack`? `WorkspaceStack`? `PulumiStack`? cc @EvanBoyle

Reply:
ah something like that could work. Keeping in mind that there is static methods on it so like: `AutomationStack.SelectAsync(..)`, `WorkspaceStack.SelectAsync(...)`, or `PulumiStack.SelectAsync(...)`

Reply:
Having returned to this I think of the 3 suggested here that I prefer `WorkspaceStack`

Reply:
I agree